core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournal.java

```diff
@@ -81,7 +81,7 @@
    * @param location the location for this journal
    */
   public UfsJournal(URI location) {
-    this(location, UnderFileSystem.Factory.get(location.toString()));
+    this(location, UnderFileSystem.Factory.get(location));
```

Comment:
correct. as it requires more refactoring to pass the ufs manager to journal related classes. 
I would defer the refactoring as this PR is already getting too big.
I think to keep the code clean, we should probably still do this, but I would prefer to do so in followup PR.